conjure-go-client/httpclient/http2_client_test.go

```diff
@@ -128,6 +133,18 @@ func newProxyServer(t *testing.T, proxyURL string) *proxyServer {
 	}
 }
 
+func (p *proxyServer) serve(t *testing.T, stopCh chan struct{}, expectErr bool) {
+	conn, err := p.ln.Accept()
+	if expectErr {
+		require.Error(t, err)
+		require.True(t, strings.Contains(err.Error(), "use of closed network connection"))
+		return
```

Comment:
@bmoylan I believe this missing return was the reason for the flaky test since the first test case expects a failure here due to the "use of closed network" error, but we continue on to try and dial the test server which caused the panic. Now we'll just return after getting an Accept error and bail out.